### DISCUSSION TITLE
提出的图片向量中有负值，查询出的距离很大

### DISCUSSION THREAD

**Discussion by ruixiu** [milvus-io/milvus#21889] [Q&A and General discussion] [answered]
[aa.txt](https://github.com/milvus-io/milvus/files/10543211/aa.txt)
![aa](https://user-images.githubusercontent.com/9940354/215676662-5e85ce6d-3774-45bc-baf0-4bc6a502f7fb.png)

[114.txt](https://github.com/milvus-io/milvus/files/10543210/114.txt)
![114](https://user-images.githubusercontent.com/9940354/215676629-7564655c-9e66-43e4-85e9-2f9e54520d29.jpg)

![result](https://user-images.githubusercontent.com/9940354/215676694-8fe28fc1-77c3-48c1-bed9-53b2801a449f.jpg)

**Comment by ruixiu** [comment, marked as answer]
原因找到了：faiss计算欧式距离（L2）的时候没有开平方

**Reply by yhmo** [reply]
> 原因找到了：faiss计算欧式距离（L2）的时候没有开平方

不开平方是因为：
1. 开方运算比较费时
2. milvus返回的结果只需要按距离大小给出最相似的结果即可，不开方也能比较出距离的大小，对结果没有影响。用户如果需要，可以自己在客户端开方。
